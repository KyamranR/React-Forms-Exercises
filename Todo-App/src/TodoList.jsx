import { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

function TodoList() {
  const [todos, setTodos] = useState([]);

  const addTodo = (newTodo) => {
    setTodos((todos) => [...todos, { id: Date.now(), task: newTodo }]);
  };

  const removeTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };

  const updateTodo = (id, newTask) => {
    setTodos((todos) =>
      todos.map((todo) => (todo.id === id ? { ...todo, task: newTask } : todo))
    );
  };

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul>
        {todos.map((todo) => (
          <Todo
            key={todo.id}
            id={todo.id}
            task={todo.task}
            removeTodo={removeTodo}
            updateTodo={updateTodo}
          />
        ))}
      </ul>
    </div>
  );
}

export default TodoList;
