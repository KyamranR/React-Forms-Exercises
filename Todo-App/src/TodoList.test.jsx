import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

test("without crashing", () => {
  render(<TodoList />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test("can add and remove todo", () => {
  const { getByText, getByPlaceholderText, queryByText } = render(<TodoList />);
  const input = getByPlaceholderText("New task");
  const button = getByText("Add Todo");

  fireEvent.change(input, { target: { value: "Test Todo" } });
  fireEvent.click(button);
  expect(getByText("Test Todo")).toBeInTheDocument();

  fireEvent.click(getByText("X"));
  expect(queryByText("Test Todo")).not.toBeInTheDocument();
});
