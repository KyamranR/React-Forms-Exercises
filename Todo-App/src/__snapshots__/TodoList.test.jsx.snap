// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        placeholder="New task"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
